python/ray/util/sgd/v2/tests/test_tune.py

```diff
@@ -127,6 +131,67 @@ def train_func(config):
         tune.run(TestTrainable)
 
 
+def test_tune_checkpoint(ray_start_2_cpus):
+    def train_func():
+        for i in range(10):
+            sgd.report(test=i)
+        sgd.save_checkpoint(hello="world")
+
+    trainer = Trainer(TestConfig())
+    TestTrainable = trainer.to_tune_trainable(train_func)
+
+    [trial] = tune.run(TestTrainable).trials
+    assert os.path.exists(
+        os.path.join(trial.checkpoint.value, TUNE_CHECKPOINT_FILE_NAME))
+
+
+def test_reuse_checkpoint(ray_start_2_cpus):
+    def train_func(config):
+        itr = 0
+        ckpt = sgd.load_checkpoint()
+        if ckpt is not None:
+            itr = ckpt["iter"] + 1
+
+        for i in range(itr, config["max_iter"]):
+            sgd.report(test=i, training_iteration=i)
+            sgd.save_checkpoint(iter=i)
```

Comment:
This looks good though I'm wondering if this ends up being unintuitive due to Tune's odd checkpointing behavior.